src/storage/scripting.cc

```diff
@@ -848,6 +853,29 @@ int RedisReturnSingleFieldTable(lua_State *lua, const char *field) {
   return 1;
 }
 
+int RedisSetResp(lua_State *lua) {
+  auto srv = GetServer(lua);
+  auto conn = srv->GetCurrentConnection();
+  int argc = lua_gettop(lua);
+
+  if (argc != 1) {
+    PushError(lua, "redis.setresp() requires one argument.");
+    return 1;
+  }
+
+  int resp = static_cast<int>(lua_tonumber(lua, -1));
+  if (resp != 2 && resp != 3) {
+    PushError(lua, "RESP version must be 2 or 3.");
+    return 1;
+  }
+  conn->SetProtocolVersion(resp == 2 ? redis::RESP::v2 : redis::RESP::v3);
+  if (resp == 3 && !srv->GetConfig()->resp3_enabled) {
+    PushError(lua, "you need set resp_enabled to yes to enable RESP3.");
```

Comment:
It looks like there is no way we can use resp2 and resp3 at the same time?
For example, use resp2 for one lua and resp3 for another lua.

```suggestion
    PushError(lua, "You need set resp3-enabled to yes to enable RESP3.");
```

Reply:
@enjoy-binbin Yes, it will support the RESP2 and RESP3 at them same time while the `resp3-enabled` was enabled. We now disable the RESP3 by default for stable consideration, so we need to check if the RESP3 was enabled here.